qiskit/result/result.py

```diff
@@ -181,6 +182,87 @@ def get_counts(self, experiment=None):
         except KeyError:
             raise QiskitError('No counts for experiment "{0}"'.format(experiment))
 
+    def get_assertion(self, experiment=None):
+        """
+        Calculate and collect results of statistical tests for an experiment.
+
+        Args:
+            experiment (QuantumCircuit): a breakpoint
+
+        Returns:
+            tuple: tuple containing:
+
+                chisq(float): the chi-squared value
+                pval(float): the p-value
+                passed(Boolean): if assertion passed
+        """
+        exp = experiment
+        counts = self.get_counts(exp)
+        assertion = exp.data[-1][0]
+
+        cbits = assertion._cbit
+        cbits = Asserts.clbits2idxs(cbits, exp)
+
+        new_counts = {}
+        for (key, value) in counts.items():
+            new_key = key.replace(' ', '')
+            new_key = ''.join([new_key[-1*(cbit+1)] for cbit in cbits][::-1])
+            new_counts.setdefault(new_key, 0)
+            new_counts[new_key] += value
+        counts = new_counts
+
+        chisq, pval, passed = assertion.stat_test(counts)
+        return chisq, pval, passed if not assertion._negate else not passed
```

Comment:
Would it be better to return an `AssertionResult` instance instead of a tuple?

Reply:
@yipenghuang0302

Reply:
(https://github.com/edasgupta/qiskit-terra/commit/3bd8cb5725605232238dfffee1bf4f6cc972662b)
now get_assertion gets the assertion object, and this method was renamed to get_assertion_stats